cypress/integration/console/devices/onboarding/qr-scan.spec.js

```diff
@@ -0,0 +1,125 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { generateHexValue } from '../../../../support/utils'
+
+describe('End device QR scan', () => {
+  const user = {
+    ids: { user_id: 'create-manual-test-user' },
+    primary_email_address: 'create-manual-test-user@example.com',
+    password: 'ABCDefg123!',
+    password_confirm: 'ABCDefg123!',
+  }
+
+  before(() => {
+    cy.dropAndSeedDatabase()
+    cy.createUser(user)
+  })
+
+  describe('QRScan', () => {
+    const appId = 'qr-test-application'
+    const application = {
+      ids: { application_id: appId },
+    }
+    const device = {
+      join_eui: '00 00 00 00 00 00 00 01',
+      dev_eui: '00 00 00 00 00 00 00 02',
+    }
+
+    const selectDevice = ({ model_id, hw_version, fw_version, band_id }) => {
+      cy.findByLabelText('Model').selectOption(model_id)
+      cy.findByLabelText('Hardware Ver.').selectOption(hw_version)
+      cy.findByLabelText('Firmware Ver.').selectOption(fw_version)
+      cy.findByLabelText('Profile (Region)').selectOption(band_id)
+    }
+
+    before(() => {
+      cy.createApplication(application, user.ids.user_id)
+    })
+
+    beforeEach(() => {
+      cy.fixture('console/devices/repository/test-brand-otaa-model3.template.json').then(
+        templateJson => {
+          cy.intercept(
+            'GET',
+            `/api/v3/dr/applications/${appId}/brands/test-brand-otaa/models/test-model3/1.0.1/EU_863_870/template`,
+            templateJson,
+          )
+        },
+      )
+      cy.fixture('console/devices/repository/test-brand-otaa.models.json').then(modelsJson => {
+        cy.intercept(
+          'GET',
+          `/api/v3/dr/applications/${appId}/brands/test-brand-otaa/models*`,
+          modelsJson,
+        )
+      })
+      cy.fixture('console/devices/repository/brands.json').then(brandsJson => {
+        cy.intercept('GET', `/api/v3/dr/applications/${appId}/brands*`, brandsJson)
+      })
+
+      cy.loginConsole({ user_id: user.ids.user_id, password: user.password })
+      cy.visit(`${Cypress.config('consoleRootPath')}/applications/${appId}/devices/add`)
+      cy.findByText('Scan end device code').click()
+    })
+
+    it('succeeds registering a device via a qr code', () => {
+      cy.findByTestId('full-error-view').should('not.exist')
+      cy.findByTestId('error-notification').should('not.exist')
+
+      cy.findByText('Scan end device QR code').should('be.visible')
+      cy.findByText('Please scan the QR code to continue.').should('exist')
```

Comment:
```suggestion
      cy.findByText('Please scan the QR code to continue.').should('to.be.visible')
```
Please always test for visibility rather than existence whenever possible.